lib/html.js

```diff
@@ -490,15 +490,31 @@ var appendDOM = function(elem, child) {
       }
 
       // in IE must explicitly nest TRs in TBODY
-      var childTag = child.tagName.toLowerCase();// child tagName
+      var childTag = child.tagName.toLowerCase();
       if (childTag && childTag !== 'tbody' && childTag !== 'thead') {
-        // insert in last tbody
-        var tBody = elem.tBodies.length > 0 ? elem.tBodies[elem.tBodies.length-1] : null;
-        if (!tBody) {
-          tBody = createElement(childTag === 'th' ? 'thead' : 'tbody');
-          elem.appendChild(tBody);
+        var isHeader = childTag === 'th';
+        // insert in existing tHead or last tBody
+        var section = isHeader ? elem.tHead : elem.tBodies[elem.tBodies.length - 1];
+        if (!section) {
+          section = createElement(isHeader ? 'thead' : 'tbody');
+          elem.appendChild(section);
+        }
+        if (childTag === 'tr') {
+          section.appendChild(child);
+        } else {
+          // find last tr
+          var tr;
+          for (var i = section.childNodes.length - 1; i >= 0 && !tr; i--) {
```

Comment:
Some of this nested logic would be more readable if it were wrapped in a method.

Reply:
I agree. I was avoiding any sort of refactoring but maybe I ought to?